src/rdkafka_partition.c

```diff
@@ -4479,3 +4499,149 @@ const char *rd_kafka_fetch_pos2str(const rd_kafka_fetch_pos_t fetchpos) {
 
         return ret[idx];
 }
+
+typedef RD_MAP_TYPE(const rd_kafka_topic_partition_t *,
+                    void *) map_toppar_void_t;
+
+/**
+ * @brief Calculates \p a ∩ \p b.
+ *        Ordered following \p a order. Elements are copied from \p a.
+ */
+static rd_kafka_topic_partition_list_t *
+rd_kafka_topic_partition_list_intersection0(
```

Comment:
I think we should have these functions at `rd_list` level. We can discuss more offline on this.

Reply:
`rd_kafka_topic_partition_list_t `is different from `rd_list `in that it only contains partitions, here I needed this functions for `rd_kafka_topic_partition_list_t` only.

Reply:
As discussed, we can add this later.